bindings/python/src/lib.rs

```diff
@@ -368,6 +368,34 @@ impl EntryMode {
     }
 }
 
+#[pyclass(module = "opendal")]
+struct PresignedRequest(od::raw::PresignedRequest);
+
+#[pymethods]
+impl PresignedRequest {
+    /// Return the URL of this request.
+    #[getter]
+    pub fn url(&self) -> String {
+        self.0.uri().to_string()
+    }
+
+    /// Return the HTTP method of this request.
+    #[getter]
+    pub fn method(&self) -> &str {
+        self.0.method().as_str()
+    }
+
+    /// Return the HTTP headers of this request.
+    #[getter]
+    pub fn headers<'p>(&'p self, py: Python<'p>) -> HashMap<&'p str, &'p PyBytes> {
```

Comment:
Convert `HeaderMap` to `HashMap` will only preserve the last value of a multi-value header. Will it become a problem?

See https://docs.rs/http/0.2.9/http/header/struct.HeaderMap.html#method.iter

Reply:
It's better to return a `HashMap<&'p str, &'p str>` here. We should raise exception when:

- HeaderMap has a multi-value header (should not happen for opendal generated presign request).
- HeaderMap has an invliad header value (It's required that all header value must be encoded).

Reply:
complete in new commit